apps/kyb-app/src/pages/Root/Root.tsx

```diff
@@ -0,0 +1,28 @@
+import { useAccessToken } from '@/common/providers/AccessTokenProvider';
+import { useEffect } from 'react';
+import { Outlet, useNavigate } from 'react-router-dom';
+import { useIsSignupRequired } from './hooks/useIsSignupRequired';
+
+export const Root = () => {
+  const { isLoading, isSignupRequired } = useIsSignupRequired();
+  const navigate = useNavigate();
+  const { accessToken } = useAccessToken();
+
+  useEffect(() => {
+    if (isLoading) return;
```

Comment:
why inside the useEffect and not outside?

Reply:
If you talking about redirect then this condition won't ever render outlet and there will be blank screen.

Reply:
Usually we have the `isLoading` checks not inside `useEffect`, can you just check if it can be pulled out?